application_sdk/outputs/parquet.py

```diff
@@ -203,69 +200,89 @@ async def write_dataframe(self, dataframe: "pd.DataFrame"):
                 name="parquet_write_errors",
                 value=1,
                 metric_type=MetricType.COUNTER,
-                labels={"type": "pandas", "mode": self.write_mode, "error": str(e)},
+                labels={"type": "pandas", "mode": "append", "error": str(e)},
                 description="Number of errors while writing to Parquet files",
             )
             logger.error(f"Error writing pandas dataframe to parquet: {str(e)}")
             raise
 
-    async def write_daft_dataframe(self, dataframe: "daft.DataFrame"):  # noqa: F821
+    async def write_daft_dataframe(
+        self,
+        dataframe: "daft.DataFrame",  # noqa: F821
+        partition_cols: Optional[List] = None,
+        write_mode: Literal["append", "overwrite", "overwrite-partitions"] = "append",
+    ):
         """Write a daft DataFrame to Parquet files and upload to object store.
 
+        Uses Daft's native file size management to automatically split large DataFrames
+        into multiple parquet files based on the configured target file size. Supports
+        Hive partitioning for efficient data organization.
+
         Args:
             dataframe (daft.DataFrame): The DataFrame to write.
+            partition_cols (Optional[List]): Column names or expressions to use for Hive partitioning.
+                Can be strings (column names) or daft column expressions. If None (default), no partitioning is applied.
+            write_mode (Literal["append", "overwrite", "overwrite-partitions"]): Write mode for parquet files ('overwrite', 'append', etc.).
+                If provided, overrides the instance write_mode.
+
+        Note:
+            - Daft automatically handles file chunking based on parquet_target_filesize
+            - Multiple files will be created if DataFrame exceeds DAPR limit
+            - If partition_cols is set, creates a Hive-partitioned directory structure
         """
         try:
+            import daft
+
             row_count = dataframe.count_rows()
             if row_count == 0:
                 return
 
+            # Use Daft's execution context for temporary configuration
+            with daft.execution_config_ctx(
+                parquet_target_filesize=self.max_file_size_bytes
+            ):
+                # Daft automatically handles file splitting and naming
+                dataframe.write_parquet(
+                    root_dir=self.output_path,
+                    write_mode=write_mode,
+                    partition_cols=partition_cols if partition_cols else [],
+                )
+
             # Update counters
             self.chunk_count += 1
             self.total_record_count += row_count
 
-            # Generate file path using path_gen function
-            if self.start_marker and self.end_marker:
```

Comment:
why was this removed? this was added by @abhishekagrawal-atlan as part of extraction for miners I believe